### DISCUSSION TITLE
Software SPI vs Hardware SPI with ESP32

### DISCUSSION THREAD

**Discussion by mottibz** [Bodmer/TFT_eSPI#889] [Q&A - General] [answered]
Hello, my second question: I am using a 2.8" TFT via SPI that uses the ILI9341 (240 X 320) with an integrated touch panel. I am connecting it to an ESP32. Using the display works fine. I am asking this as I did not find good explanation and how-to guide. I connected the wires using the library default pin numbers (as defined in the user_setup.h include file for ESP32). I understand that this uses the Software SPI mode. (1) What are the benefits of using the Hardware SPI mode? (2) How do I set this up for the ESP32 and used pins, and library modifications?
Thanks a lot, Motti

**Comment by Bodmer** [comment, marked as answer]
The library uses a hardware based SPI, this gives better performance.
